src/kde_analysis/eval_m1m2chieff_poiss_weights.py

```diff
@@ -206,79 +492,200 @@ def get_rate_m_chieff2D(m1_query, m2_query, Rate):
+            'cfgrid': cfgrid,
+            'alpha': alpha,
+            'group': group,
+            'apply_symmetry': True,
+            'dimension_names': ['m1', 'm2', 'chieff']
+        }
+
+        # Helper function to determine constraint based on keep_dims
+        def get_constraint(keep_dims):
+            """Apply constraint unless we're keeping full m1-m2 plane"""
+            if set(keep_dims) == {0, 1}:
+                return None
+            return 'm1>=m2'
+
+        # ========== 2D: M1 vs Chieff ==========
+        keep_dims = [0, 2]
+        symmetric_factor = 1./2
+        kde_result = create_marginalized_kde(
+            samples=samples,
+            keep_dims=keep_dims,
+            apply_constraint=get_constraint(keep_dims),
+            **common_params
+        )
+        kdeM1chieff = kde_result['kde_values'] * symmetric_factor
+        rateM1chieff = compute_rate_from_kde(
+            kdeM1chieff, VT_3d,
+            weights_over_VT=weights_over_VT if vt_weights else None,
+            N=Nev,
+            vt_weights=vt_weights
+        )
 
-        train_kde = kde.VariableBwKDEPy(
-            symmetric_samples,
-            weights,
-            input_transf=('log', 'log', 'none'),
-            stdize=True,
-            rescale=[1/bwx, 1/bwy, 1/bwz],
-            bandwidth=np.tile(per_point_bandwidth, 2)
+        # ========== 2D: M2 vs Chieff ==========
+        keep_dims = [1, 2]
+        symmetric_factor = 1./2
+        kde_result = create_marginalized_kde(
+            samples=samples,
+            keep_dims=keep_dims,
+            apply_constraint=get_constraint(keep_dims),
+            **common_params
         )
-    else:
-        train_kde = akde.AdaptiveBwKDE(
-            symmetric_samples,
-            weights,
-            input_transf=('log', 'log', 'none'),
-            stdize=True,
-            rescale=[1/bwx, 1/bwy, 1/bwz],
-            alpha=alpha
+        kdeM2chieff = kde_result['kde_values']*symmetric_factor
+        rateM2chieff = compute_rate_from_kde(
+            kdeM2chieff, VT_3d,
+            weights_over_VT=weights_over_VT if vt_weights else None,
+            N=Nev,
+            vt_weights=vt_weights
         )
 
-    # Evaluate KDE
-    eval_kde3d = train_kde.evaluate_with_transf(eval_samples)
-    KDE_3d = eval_kde3d.reshape(XX.shape)
+        # ========== 2D: M1 vs M2 (no constraint) ==========
+        keep_dims = [0, 1]
+        symmetric_factor = 1.
+        kde_result = create_marginalized_kde(
+            samples=samples,
+            keep_dims=keep_dims,
+            apply_constraint=get_constraint(keep_dims),
+            **common_params
+        )
+        KDEm1m2 = kde_result['kde_values']*symmetric_factor
+        ratem1m2 = compute_rate_from_kde(
+            KDEm1m2, VT_3d,
+            weights_over_VT=weights_over_VT if vt_weights else None,
+            N=Nev,
+            vt_weights=vt_weights
+        )
+
+        # ========== 1D: M1 only ==========
```

Comment:
We discussed that it's more accurate to get the 1d mass KDEs by evaluating in 2d (integrating only over chieff) and numerically integrating over the m1>m2 region, as we did previously. 
Then can we keep that please. 

If you also want to make these 'approximate' 1d mass evaluations can you give them a different name?